splash/tests/test_execute_callbacks.py

```diff
@@ -834,6 +834,35 @@ def test_error_handled_in_callback(self):
         self.assertStatusCode(resp, 200)
         self.assertEqual(resp.text, "error")
 
+    def test_multiple_calls(self):
+        resp = self.request_lua("""
+        local treat = require('treat')
+        function main(splash)
+          local o = {false, false, false, false, false, false}
+
+          local timer1 = splash:call_later(function() o[1] = true end, 0.5)
+          local timer2 = splash:call_later(function() o[2] = true end, 0.5)
+          local timer3 = splash:call_later(function() o[3] = true end, 0.5)
+          local timer4 = splash:call_later(function() o[4] = true end, 0.5)
+          local timer5 = splash:call_later(function() o[5] = true end, 0.5)
+          local timer6 = splash:call_later(function() o[6] = true end, 0.5)
+
+          timer1:cancel()
+          timer2:cancel()
+          timer3:cancel()
+          timer4:cancel()
+          timer5:cancel()
+          timer6:cancel()
+
+          splash:wait(1)
+
+          return treat.as_array(o)
+        end
+        """)
+
+        self.assertStatusCode(resp, 200)
+        self.assertEqual(resp.json(), [False, False, False, False, False, False])
```

Comment:
If you run this test w/o this fix you will get `[False, True, True, True, True, False]`. This happens because when `setup_property_access` is called it sets `__index` and `__setindex` metamethods to the provided class. And that way only the first and the last created instances will work correctly, because for the class (table) of the first instance is created without any overrides and the last instance works because its class methods  has right `self` arguments.